gap/CatRepsWithCAP.gi

```diff
@@ -7,9 +7,13 @@
 ##
 InstallGlobalFunction( ConvertToMapOfFinSets,
    function( objects, gen )
-        local T, isb, S, G, j, i;
+        local O, T, isb, S, G, j, i;
 	
-	T := FinSet(gen);
+	for O in objects do
+		if Length(Intersection(gen,O)) > 0 then
+			T := FinSet(O);
```

Comment:
1. After that you can break.
2. After the loop you have to check `if IsBound( T )` otherwise, throw and error.